Update ReadMe with cherry-pick instrutions for terraform repo (#1767)
hashicorp/terraform-website: README.md

↥ [back to top](#table-of-contents)

### For changes in this repo

Merge the PR to master, and the site will automatically deploy in about 20m. 🙌

### For changes in `hashicorp/terraform`

> Merge the PR to master. The changes will appear in the next major Terraform release.

Reviewer: ```suggestion
Merge the PR to main. The changes will appear in the next major Terraform release.
```

If you need your changes to be deployed sooner, cherry-pick them to:
- the current release branch (e.g. `v1.0`) and push. They will be deployed in the next minor version release (once every two weeks).
- the `stable-website` branch and push. They will be included in the next site deploy (see below). Note that the release process resets `stable-website` to match the release tag, removing any additional commits. So, we recommend always cherry-picking to the version branch first and then to `stable-website` when needed.

### Deployment

> The [terraform.io][] site gets deployed by a CI job, currently managed by CircleCI. This job can be run manually by many people within HashiCorp, and also runs automatically whenever a user in the HashiCorp GitHub org merges changes to master in this repository. (Note that Terraform releases create sync commits to terraform-website, which will trigger a deploy.)

Reviewer: ```suggestion
Currently, HashiCorp uses a CircleCI job to deploy the [terraform.io](terraform.io) site. This job can be run manually by many people within HashiCorp, and also runs automatically whenever a user in the HashiCorp GitHub org merges changes to master in this repository. Note that Terraform releases create sync commits to `terraform-website`, which will trigger a deploy.

```

> New commits in `hashicorp/terraform` don't automatically deploy the site, but an unrelated site deploy will usually happen within a day. If you can't wait that long, you can do a manual CircleCI build or ask someone in the #proj-terraform-docs channel to do so:

Reviewer: ```suggestion
New commits in `hashicorp/terraform` don't automatically deploy the site, but an unrelated site deploy will usually happen within a day. If you can't wait that long, you can trigger a manual CircleCI build or ask someone in the #proj-terraform-docs channel to do so:
```
